tests/results/ResultFilterTest.py

```diff
@@ -505,3 +508,25 @@ def test_no_range(self):
         expected_diff = Diff.from_string_arrays(test_file, ["abc"])
 
         self.assertEqual(result_diff, expected_diff)
+
+    def test_removed_file(self):
+        test_file = ["abc"]
+        test_file_dict = {"test_file": test_file}
+        test_mod_file_dict = {}
+
+        ensure_files_present(test_file_dict, test_mod_file_dict)
+
+        self.assertEqual(
+            test_mod_file_dict,
+            {"test_file": []})
+
+    def test_added_file(self):
+        test_file = ["abc"]
+        test_file_dict = {}
+        test_mod_file_dict = {"test_file": test_file}
+
+        ensure_files_present(test_file_dict, test_mod_file_dict)
+
+        self.assertEqual(
+            test_file_dict,
+            {"test_file": []})
```

Comment:
please add a test that verifies that if you have a new file added with a result this result is successfully recognized as new?